autoscheduler/frontend/src/redux/reducers.ts

```diff
@@ -61,10 +102,117 @@ function courseCards(
   }
 }
 
+// sets whether the user is selecting busy time or not
+// in the future, this could also support preferred times
+function availabilityMode(
+  state: AvailabilityType = AvailabilityType.NONE, action: AvailabilityModeAction,
+): AvailabilityType {
+  if (action.type === SET_AVAILABILITY_MODE) { return action.mode; }
+  return state;
+}
+
+// stores all information about the user's availability
+function availability(
+  state: Availability[] = [], action: AvailabilityAction,
+): Availability[] {
+  switch (action.type) {
+    case ADD_AVAILABILITY:
+    case MERGE_AVAILABILITY: {
+      // check for overlaps between the availability to be added and pre-existing ones
+      let overlapsFound = 0; // counts merging overlaps only
+      let overlapIdx = -1;
+
+      const initNewAv = action.type === ADD_AVAILABILITY
+        ? argsToAvailability(action.availability)
+        : state[state.length - 1];
+      let newAv = initNewAv;
+      const newState = state.reduce<Availability[]>((avsList, oldAv, idx): Availability[] => {
+        // only counts as an overlap if they're on the same day of the week
+        if (oldAv.dayOfWeek === newAv.dayOfWeek) {
+          const avWithEarlierStart = getStart(oldAv) < getStart(newAv) ? oldAv : newAv;
+          const avWithLaterStart = avWithEarlierStart === oldAv ? newAv : oldAv;
+          // overlap detected
+          if (getEnd(avWithEarlierStart) >= getStart(avWithLaterStart)) {
+            // if they're the same type, merge them
+            if (oldAv.available === newAv.available) {
+              overlapsFound += 1;
+              const oldOverlapIdx = overlapIdx;
+              overlapIdx = idx;
+              const newEnd = Math.max(getEnd(oldAv), getEnd(newAv));
+              const newEndHrs = Math.floor(newEnd / 60);
+              const newEndMins = newEnd % 60;
+              newAv = {
+                available: oldAv.available,
+                dayOfWeek: oldAv.dayOfWeek,
+                startTimeHours: avWithEarlierStart.startTimeHours,
+                startTimeMinutes: avWithEarlierStart.startTimeMinutes,
+                endTimeHours: newEndHrs,
+                endTimeMinutes: newEndMins,
+              };
+              if (overlapsFound === 2) {
+                avsList.splice(oldOverlapIdx, 1);
+                return avsList;
+              }
+              return avsList.filter((av) => av !== oldAv);
```

Comment:
I'd also say add comments to this block just explaining the difference between returning in the `if` statement versus the default `.filter` return

Reply:
There is no difference. They have both been replaced with `return avsList;`